cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -126,6 +126,26 @@ impl RexFlags {
     }
 }
 
+/// Generate the proper Rex flags for the given operand size.
+impl From<OperandSize> for RexFlags {
+    fn from(size: OperandSize) -> Self {
+        match size {
+            OperandSize::Size64 => RexFlags::set_w(),
+            _ => RexFlags::clear_w(),
+        }
+    }
+}
+/// Generate Rex flags for an OperandSize/register tuple.
+impl From<(OperandSize, u8)> for RexFlags {
```

Comment:
It might be a little cleaner / more type-safe to take an `(OperandSIze, Reg)` tuple here (and use `int_reg_enc()`), I think.

Reply:
Done.

For consistency I also changed always_emit_if_8bit_needed to also take a Reg as its argument.